### PR TITLE
Fixed AsyncEffect.java clear local variables

### PR DESCRIPTION
**Pull request by SkyCraft78**
Hi!
As said in the title, this small PR fixes the clear of local vars into an async effect.

**Example code used:**
```applescript
command skent:
	trigger:
		set {_filepath} to "plugins/example/hello.txt" 
		send "%{_filepath}%"
		create file path "%{_filepath}%"
		send "%{_filepath}%"
```

Where `{_filepath}` were `<none>` after the file creation.

### CODE REVIEW

File: src/main/resources/plugin.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: Skent
 description: Manage files with Skript.
 author: Olyno
-version: 3.0.0
```
Comment (Olyno) [inline src/main/resources/plugin.yml:4]: Change not related to the pull request.

File: src/main/java/com/olyno/skent/util/skript/AsyncEffect.java [hunk 2]
```diff
@@ -37,20 +37,20 @@ public boolean init(Expression<?>[] expr, int matchedPattern, Kleenean isDelayed
     protected void execute(Event e) {
         this.event = e;
         localVars = Variables.removeLocals(e);
+        if (localVars != null)
+            Variables.setLocalVariables(e, localVars);
```
Comment (Olyno) [inline src/main/java/com/olyno/skent/util/skript/AsyncEffect.java:41]: Please respect format of code, use brackets.

File: src/main/java/com/olyno/skent/util/skript/AsyncEffect.java [hunk 3]
```diff
@@ -37,20 +37,20 @@ public boolean init(Expression<?>[] expr, int matchedPattern, Kleenean isDelayed
...
-        if (localVars != null) {
+        if (localVars != null)
             Variables.setLocalVariables(this.event, localVars);
-        }
         this.executeAsync(this.event);
-        if (this.needExecuteCode) {
+        if (this.needExecuteCode)
             this.runSection(this.event);
-        }
         Variables.removeLocals(this.event);
```
Comment (Olyno) [inline src/main/java/com/olyno/skent/util/skript/AsyncEffect.java:54]: Not related to the fix.

**Review by Olyno — CHANGES_REQUESTED** [review]
Hi, thanks you for your contribution! This is a fix that I didn't be able to fix due to my poor time. But before merge the pull request, please revert unecesary changes.

**Comment by Olyno** [comment]
After testing more, this pull request doesn't fix all async problem.
```py
command test:
    trigger:
        set {_dir} to "plugins/skenttest/tests"
        create dir path {_dir}:
            broadcast "Dir: %{_dir}%" # <none>
            create file path "%{_dir}%/test.txt":
                broadcast "Dir: %{_dir}%" # <none>
                broadcast "Creation done!"
        broadcast "Dir: %{_dir}%" # "plugins/skenttest/tests"
        broadcast "creating!"
```

**Comment by SkyCraft78** [comment]
Hi!
I just fixed the unnecessary changed, sorry I didn't know about them 😅 
I also debugged a little bit more your example code, and look likes only the second dir test is `<none>`:
```
command test:
    trigger:
        set {_dir} to "plugins/skenttest/tests"
        create dir path {_dir}:
            broadcast "Dir 1: %{_dir}%" # <none>
            create file path "%{_dir}%/test.txt":
                broadcast "Dir 2: %{_dir}%" # <none>
                broadcast "Creation done!"
        broadcast "Dir 3: %{_dir}%" # "plugins/skenttest/tests"
        broadcast "creating!"
```
I am getting:
```
Dir 3: plugins/skenttest/tests
Dir 1: plugins/skenttest/tests
creating!
Dir 2: <none>
Creation done!
```
So the variable is cleared only in the second scope, which is pretty weird because the code execution is the same 😂
Nevermind, I'll take a deeper look right now and tell you what did I found ^^

**Comment by SkyCraft78** [comment]
Here you go, you were removing the variables after the section execution, so the next section wasn't able to store them x)

**Comment by Olyno** [comment]
Which versions of server and Skript are you using? I still have the issue with latest Skript stable version (2.5.3) and Tuinity 1.16.5

**Comment by SkyCraft78** [comment]
I got Skript v2.5.3, Paper v1.12 (git-paper-1618) and the build of Skent i made with the current commits
Just to be sure we are on the same way x) Using that code, with the latest build (check joined files):
```
command test:
    trigger:
        set {_dir} to "plugins/skenttest/tests"
        create dir path {_dir}:
            broadcast "Dir 1: %{_dir}%" # "plugins/skenttest/tests"
            create file path "%{_dir}%/test.txt":
                broadcast "Dir 2: %{_dir}%" # "plugins/skenttest/tests"
                broadcast "Creation done!"
        broadcast "Dir 3: %{_dir}%" # "plugins/skenttest/tests"
        broadcast "creating!"
```
It sout me:
image
(PS: I send a discord URL since github doesn't support .jar files xD)
https://cdn.discordapp.com/attachments/727897831133675540/862291772268609546/Skent.jar

**Comment by Olyno** [comment]
Tested with Paper, Tuinity, your pull request and your build, it worked 2 times in a row and then stopped to work anymore. It's not because it worked 1 time with some luck it fix the issue. Try to upgrade your minecraft server to latest to make your tests, and execute multiple times the command to test it.

**Comment by Olyno** [comment]
Closing due to inactivity
